src/jsonrpc/api.rs

```diff
@@ -258,6 +282,14 @@ pub fn handle_request(control: &DaemonControl, req: Request) -> Result<Response,
         }
         "getinfo" => serde_json::json!(&control.get_info()),
         "getnewaddress" => serde_json::json!(&control.get_new_address()),
+        "listaddresses" => {
+            let params = req.params.ok_or_else(|| {
```

Comment:
As the parameters are optional, I'm wondering if you could keep `params` as `Option<Params>` and pass that to `list_addresses`.

I'm able to run `listaddresses` without any parameters so I guess `req.params` is `Some` also in that case and it works fine, but not sure if there might be a situation where it is `None`.

@darosior may be able to help.

Reply:
you seems right, i wonder why `Error::invalid_params` is not triggered when call listaddresses w/o args? shouldn't it in that case?

Reply:
That's most likely you tested with the CLI which always includes a `params` entry in the JSONRPC request. But that's not mandated by the [JSONRPC2 specs](https://www.jsonrpc.org/specification#request_object).

I agree with @jp1ac4 this shouldn't error if params is None. You can take inspiration from how `list_coins` does this.